### DISCUSSION TITLE
Unable to run sample example/http/http.go - apikey.json

### DISCUSSION THREAD

**Discussion by aleblanc70** [zitadel/zitadel#6839] [Q&A]
Ok I do not get it...
from the sample : https://github.com/zitadel/zitadel-go/blob/main/example/http/http.go

introspection, err := http_mw.NewIntrospectionInterceptor(*issuer, "apikey.json")

What do I put in my apikey.json
I tried ClientId and ClientSecret nothing works...

When I debug I fall on: 
return NewResourceServerJWTProfile(issuer, c.ClientID, c.KeyID, []byte(c.Key), options...)

What will be c.KeyID and c.Key ?

I was not able to run any samples :(. 
Certainly a novice error!!!

**Comment by hifabienne** [comment]
hei @aleblanc70 
When you create a service user in ZITADEL.
You do have a tab Keys.
There you can create a new key and you get a json file to download
<img width="1313" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/a188ae18-9a85-4527-9207-7c04688ddf0d">

**Comment by aleblanc70** [comment]
okay I was able to generate the file. 
Now using Insomia to call the /protected. I got invalid token. 

I tried generate the file as jwt and bearer.
In my bearer token I tried to pass my  keyid or userid.

Always invalid token! :(

**Comment by hifabienne** [comment]
Can you describe a little bit more in detail what you are trying to do?
Maybe also send a curl?
So which endpoint do you try to call?
Do you need a service account? Or a human user?

**Reply by aleblanc70** [reply]
I want to understand the sample: https://github.com/zitadel/zitadel-go/blob/main/example/http/http.go
see below

**Comment by aleblanc70** [comment]
Yes I try to run the sample : 

https://github.com/zitadel/zitadel-go/blob/main/example/http/http.go
Line 26: is runing ok
Line 27:  I cannot run it, it give me invalid token.

Insomia is a tool like postman.  

With Curl what will be the way to call http://localhost:5001/protected like in the sample.

Cause with Insomia I assume I do not send the correct identifier in the header...
I do not know how to send it with curl, can you give me a sample please ?

Where do I get a valid token here:
curl -i -H "Authorization: Bearer ${token}" localhost:5001/protected

KeyId, UserId and Id do not work :(

**Comment by hifabienne** [comment]
@stebenz @livio-a Can you help here?
